update_manifest.py

```diff
@@ -1,29 +1,85 @@
 import hashlib
-import os
+import logging
+import pathlib
+from typing import Optional
 import xml.etree.ElementTree
 
+"""This script requires at least Python 3.7.0 to run."""
 
-def update_manifest():
-    manifest = xml.etree.ElementTree.parse("manifest.xml")
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.NullHandler())
+
+
+def update_manifest(version: Optional[str] = None, replace: bool = False):
+    """Update SHA1 hashes and version number for Path of Building's manifest file.
+
+    :param version: Three-part version number following https://semver.org/.
+    :param replace: Whether to overwrite the existing manifest file.
+    :return:
+    """
+    supported_extensions = {".dll", ".jpg", ".lua", ".md", ".png", ".txt"}
```

Comment:
Can we add `.tga` and `.tgf` extensions to this list?  I don't anticipate updates to those files, but I don't want surprises in case we do.  Or do we still need/want to restrict filetypes at all?

Reply:
Actually, once we ship a Launcher update, we'll need `.exe` as well, so I think we can do away with the `supported_extensions`

Reply:
I don't know why the initial version does that, actually. I just kept it in.

Reply:
I'll remove it, agreed.
And just fyi, in Python, that's a set literal. Lists have square brackets.